.gitlab-ci.yml

```diff
@@ -0,0 +1,78 @@
+  script:
+    # upload artifacts
+    - su -c "$MVN -B -Pdocs-and-source -Dartifactory_user=${SVC_ANTIFRAUD_CI_USER} -Dartifactory_pw=${SVC_ANTIFRAUD_CI_PWD} -Dfindbugs.skip -Dcheckstyle.skip=true -Denforcer.skip=true -Drat.skip=true -Drat.ignoreErrors=true -DskipTests -DaltDeploymentRepository=releases.repo.splunk.com::default::https://repo.splunk.com/artifactory/maven-splunk-local --settings $DEPLOY_SETTINGS -DretryFailedDeploymentCount=10 -DdeployAtEnd=true deploy"
+    # this will upload the release tarball jar. The jar will be uploaded to the artifactId folder, eg
+    #   https://repo.splunk.com/artifactory/maven-splunk-local/org/apache/flink/flink-release/<version>/
+    - su -c "$MVN -B -Durl=https://repo.splunk.com/artifactory/maven-splunk-local -DrepositoryId=releases.repo.splunk.com -DgroupId=org.apache.flink -DartifactId=flink-release -Dversion=${TARBALL_VERSION} -Dpackaging=tar.gz -DgeneratePom=true -Dfile=build-target/deploy/${TARBALL_VERSION}.tgz --settings $DEPLOY_SETTINGS -Dartifactory_user=${SVC_ANTIFRAUD_CI_USER} -Dartifactory_pw=${SVC_ANTIFRAUD_CI_PWD} -DdeployAtEnd=true deploy:deploy-file"
+  only:
+    - /^release-.*-splunk.*$/
+
+upload:test:
+  stage: upload
+  dependencies:
+    - build:artifacts
+  script:
+    # upload artifacts
+    - su -c "$MVN -B -Pdocs-and-source -Dartifactory_user=${SVC_ANTIFRAUD_CI_USER} -Dartifactory_pw=${SVC_ANTIFRAUD_CI_PWD} -Dfindbugs.skip -Dcheckstyle.skip=true -Denforcer.skip=true -Drat.skip=true -Drat.ignoreErrors=true -DskipTests -DuniqueVersion=false -DaltDeploymentRepository=releases.repo.splunk.com::default::https://repo.splunk.com/artifactory/maven-splunk-local --settings $DEPLOY_SETTINGS -DretryFailedDeploymentCount=10 -DdeployAtEnd=true deploy"
+    # this will upload the release tarball jar. The jar will be uploaded to the artifactId folder, eg
+    #   https://repo.splunk.com/artifactory/maven-splunk-local/org/apache/flink/flink-release/<version>/
+    - su -c "$MVN -B -Durl=https://repo.splunk.com/artifactory/maven-splunk-local -DrepositoryId=releases.repo.splunk.com -DgroupId=org.apache.flink -DartifactId=flink-release -Dversion=${TARBALL_VERSION} -Dpackaging=tar.gz -DgeneratePom=true -Dfile=build-target/deploy/${TARBALL_VERSION}.tgz -DuniqueVersion=false --settings $DEPLOY_SETTINGS -Dartifactory_user=${SVC_ANTIFRAUD_CI_USER} -Dartifactory_pw=${SVC_ANTIFRAUD_CI_PWD} -DdeployAtEnd=true deploy:deploy-file"
+  except:
+    - /^release-.*-splunk.*$/
+
+upload:external:
+  stage: upload
+  script:
+    # Upload to external repo
+    - su -c "$MVN -B -Pdocs-and-source -Dartifactory_user=${ARTIFACTORY_USER} -Dartifactory_pw=${ARTIFACTORY_PASS} -Dfindbugs.skip -Dcheckstyle.skip=true -Denforcer.skip=true -Drat.skip=true -Drat.ignoreErrors=true -DskipTests -DaltDeploymentRepository=releases.repo.splunk.com.external::default::https://repo.splunk.com/artifactory/maven-splunk-release --settings $DEPLOY_SETTINGS -DretryFailedDeploymentCount=10 -DdeployAtEnd=true deploy"
```

Comment:
Are we missing a step here for uploading the tarball jar to external repo?  Looking at the logs for the branch https://cd.splunkdev.com/data-availability/flink-mirror/-/jobs/21983330/raw; `upload:external` would also require a similar step as the second one is doing the uploading to external repo

Reply:
The tarball is not required for the external upload since the only requirement for the external repository is the transitive  dependencies on flink artifacts. The 1.9 repo does the upload similarly as well because of this reason I believe

Reply:
gotcha! ty